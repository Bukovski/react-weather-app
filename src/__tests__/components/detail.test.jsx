import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import Detail from "../../components/detail.component";


describe("Detail Component", () => {
	let data;
	
	beforeAll(() => {
		data = {
			"weatherDescription": "BROKEN CLOUDS",
			"weatherIcon": "wi wi-day-cloudy",
			"windSpeed": 2.05,
			"humidity": "59",
			"pressure": "1022"
		};
	})
	
	it('should render props', async () => {
		const { container } = render(<Detail data={ data }/>);
		
		await waitFor(() => {
			expect(screen.queryByText('59')).toBeInTheDocument();
			expect(screen.queryByText('1022')).toBeInTheDocument();
			expect(screen.queryByText('2.05')).toBeInTheDocument();
			expect(screen.queryByText('BROKEN CLOUDS')).toBeInTheDocument();
		}, { timeout: 1500 });
		
		expect(container.firstChild).toMatchSnapshot();
	});
})