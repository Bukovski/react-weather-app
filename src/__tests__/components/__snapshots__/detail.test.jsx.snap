// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail Component should render props 1`] = `
<div
  class="detail"
>
  <div
    class="detail__wrap"
  >
    <span
      class="detail__icon"
    >
      <i
        class="wi wi-day-cloudy"
      />
    </span>
    <div
      class="detail__description"
    >
      BROKEN CLOUDS
    </div>
  </div>
  <div
    class="detail__box"
  >
    <div
      class="detail__indicator"
    >
      <div
        class="detail__info"
      >
        <span>
          2.05
        </span>
        <span
          class="detail__measure"
        >
          m/s
        </span>
      </div>
      <div
        class="detail__name"
      >
        wind speed
      </div>
    </div>
    <div
      class="detail__indicator"
    >
      <div
        class="detail__info"
      >
        <span>
          59
        </span>
        <span
          class="detail__measure"
        >
          %
        </span>
      </div>
      <div
        class="detail__name"
      >
        humidity
      </div>
    </div>
    <div
      class="detail__indicator"
    >
      <div
        class="detail__info"
      >
        <span>
          1022
        </span>
        <span
          class="detail__measure"
        >
          hpa
        </span>
      </div>
      <div
        class="detail__name"
      >
        pressure
      </div>
    </div>
  </div>
</div>
`;
